ecosystem/typescript/sdk/examples/typescript-esm/offer_capabilities.ts

```diff
@@ -0,0 +1,133 @@
+import { AptosAccount, FaucetClient, Network, Provider, HexString, TxnBuilderTypes, BCS, Types } from "aptos";
+import assert from "assert";
+
+const CORE_CODE_ADDRESS = new HexString("0x0000000000000000000000000000000000000000000000000000000000000001");
+const ED25519_ACCOUNT_SCHEME = 0;
+
+// Works for both SignerCapabilityOffer and RotationCapabilityOffer.
+// The structName changes and chainId isn't used in SignerCapabilityOffer
+type CapabilityOfferProofChallengeV2 = {
+  accountAddress: TxnBuilderTypes.AccountAddress;
+  moduleName: string;
+  structName: string;
+  chainId?: number; // not used in SignerCapabilityOffer
```

Comment:
nite: usually optional params are listed at the bottom - not big of a deal here but nice to follow good practices

Reply:
ah yea my bad! Forgot to re-arrange it because I needed the bytes to be in a specific order and was moving the params around without the function, will move it to the end